src/schema/ecommerce/create_order_with_artwork_mutation.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  GraphQLInputObjectType,
+  GraphQLNonNull,
+  GraphQLInt,
+  GraphQLString,
+  graphql,
+} from "graphql"
+
+import { mutationWithClientMutationId } from "graphql-relay"
+import { OrderOrFailureUnionType } from "schema/ecommerce/types/order_or_error_union"
+import gql from "test/gql"
+
+const CreateOrderInputType = new GraphQLInputObjectType({
+  name: "CreateOrderInput",
+  fields: {
+    artworkId: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "BSON ID of artwork",
+    },
+    editionSetId: {
+      type: GraphQLString,
+      description: "ID of artwork's edition set",
+    },
+    quantity: {
+      type: GraphQLInt,
+      description: "quantity of artwork",
+    },
+  },
+})
+
+export const CreateOrderWithArtworkMutation = mutationWithClientMutationId({
+  name: "CreateOrderWithArtwork",
+  description: "Creates an order with an artwork",
+  inputFields: CreateOrderInputType.getFields(),
+  outputFields: {
+    orderOrError: {
+      type: OrderOrFailureUnionType,
+    },
+  },
+  mutateAndGetPayload: (
+    { artworkId, editionSetId, quantity },
+    context,
+    { rootValue: { accessToken, exchangeSchema } }
+  ) => {
+    if (!accessToken) {
+      return new Error("You need to be signed in to perform this action")
+    }
+
+    const mutation = gql`
+      mutation createOrderWithArtwork(
+        $artworkId: String!
+        $editionSetId: String
+        $quantity: Int
+      ) {
+        ecommerce_createOrderWithArtwork(
+          input: {
+            artworkId: $artworkId
+            editionSetId: $editionSetId
+            quantity: $quantity
+          }
+        ) {
+          orderOrError {
+            ... on EcommerceOrderWithMutationSuccess {
+              order {
```

Comment:
Is this because the actual union type is defined in Exchange? It's sort of frustrating to have to list out all of the fields here :( (just one more thing that needs to be updated when we add a new one).